cli/azd/pkg/operations/printer.go

```diff
@@ -0,0 +1,120 @@
+package operations
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/messaging"
+	"github.com/google/uuid"
+)
+
+type messagePrinter struct {
+	console            input.Console
+	subscriber         messaging.Subscriber
+	subscription       *messaging.Subscription
+	currentMessage     string
+	currentOperationId uuid.UUID
+	spinnerRunning     bool
+}
+
+func NewMessagePrinter(subscriber messaging.Subscriber, console input.Console) Printer {
+	return &messagePrinter{
+		subscriber: subscriber,
+		console:    console,
+	}
+}
+
+// Starts listening for messages to print to the console
+func (p *messagePrinter) Start(ctx context.Context) error {
+	if p.subscription != nil {
+		return fmt.Errorf("printer already started")
+	}
+
+	filter := func(ctx context.Context, message *messaging.Envelope) bool {
+		return message.Type == defaultMessageKind
+	}
+
+	subscription, err := p.subscriber.Subscribe(ctx, filter, p.receiveMessage)
+	if err != nil {
+		return err
+	}
+
+	p.subscription = subscription
+	return nil
+}
+
+// Stops listening for messages
+func (p *messagePrinter) Stop(ctx context.Context) error {
+	if p.subscription == nil {
+		return fmt.Errorf("printer not started")
+	}
+
+	subscrption := p.subscription
+	p.subscription = nil
+	return subscrption.Close(ctx)
+}
+
+// Flushes any pending messages and blocks until they have all been handled
+func (p *messagePrinter) Flush(ctx context.Context) error {
+	if p.subscription == nil {
+		return fmt.Errorf("printer not started")
+	}
+
+	return p.subscription.Flush(ctx)
+}
+
+// Receives messages from the message bus and prints them to the console
+func (p *messagePrinter) receiveMessage(ctx context.Context, envelope *messaging.Envelope) {
+	msg, ok := envelope.Value.(*Message)
```

Comment:
All spinner orchestration is handled by this operation printer component.

Reply:
This `Printer` component can be used across all our commands to provide a unified mean of displaying our UX patterns in a consistent and easy to maintain component.